index.ts

```diff
@@ -176,30 +189,21 @@ wss.on("request", function (request) {
     };
   }
   const room: Room = rooms[roomId];
+  let conn: Connection = {
+    name,
+    connection,
+    id
+  };
+  room.connections.push(conn);
```

Comment:
why move this here and not keep it in the `join` case?
With the move the client effectively joins with establishing a web socket connection and before even `join`ing explicitly.

For now this is just a negligible race condition but in the future we may change the protocol such that `join`ing is not always mandatory...